src/privileged/trackers/api.js

```diff
@@ -89,6 +89,10 @@ this.trackers = class extends ExtensionAPI {
         async pageBeforeUnloadCallback(e) {
           const tabId = tabTracker.getBrowserTabId(e.target);
           try {
+            const uri = Services.io.newURI(e.data.telemetryData.completeLocation);
```

Comment:
you technically wouldn't need to pass the complete location and could just use the hostname, since permissions are based on origin (scheme + host), but this works fine :)

Reply:
ooh, I thought it could handle hostnames without the scheme, but of course, that doesn't make sense, since we need the scheme for the origin as well. Thanks, you're absolutely right!